app/coffee/DocArchiveManager.coffee

```diff
@@ -0,0 +1,65 @@
+MongoManager = require "./MongoManager"
+MongoAWS = require "./MongoAWS"
+LockManager = require "./LockManager"
+DocstoreHandler = require "./DocstoreHandler"
+logger = require "logger-sharelatex"
+_ = require "underscore"
+async = require "async"
+settings = require("settings-sharelatex")
+
+module.exports = DocArchiveManager =
+
+	archiveAllDocsChanges: (project_id, callback = (error, docs) ->) ->
+		if settings.filestore?.backend != "s3"
+			return callback(null)
+		DocstoreHandler.getAllDocs project_id, (error, docs) ->
+			if error?
+				return callback(error)
+			else if !docs?
+				return callback new Error("No docs for project #{project_id}")
+			jobs = _.map docs, (doc) ->
+				(cb)-> DocArchiveManager.archiveDocChangesWithLock project_id, doc._id, cb
+			async.series jobs, callback
+
+	archiveDocChangesWithLock: (project_id, doc_id, callback = (error) ->) ->
+		LockManager.runWithLock(
+			"HistoryArchiveLock:#{doc_id}",
```

Comment:
I find this a little hard to read personally and it relies on coffescript spacing which can be a touch unpredictable imo.

	job = (releaseLock) ->
		DocArchiveManager.archiveDocChanges project_id, doc_id, releaseLock
	LockManager.runWithLock("HistoryArchiveLock:#{doc_id}", job, callback)